src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -1853,6 +1869,58 @@ private FunctionGroup FirstFunctionGroupInInheritanceChain(RuntimeCore runtimeCo
             return ret;
         }
 
+        /// <summary>
+        /// Determine if the formalParameters are homogenious and intialize a flat list of final formalParameters
+        /// </summary>
+        /// <returns>true if the formalParameters are homogenious</returns>
+        private static bool AreParametersHomogenious(List<StackValue> formalParameters, RuntimeCore runtimeCore, List<StackValue> finalFormalParameters)
+        {
+            foreach (var formalParameter in formalParameters)
+            { 
+                //expand array if required to compare inputs
+                if (formalParameter.IsArray)
+                {
+                    DSArray array = runtimeCore.Heap.ToHeapObject<DSArray>(formalParameter);
+                    StackValue[] flatParameters = array.Values.ToArray();
+
+                    switch (flatParameters.Length)
+                    {
+                        case 0:
+                            //set function result false and exit due to empty list
+                            return false;
+                        case 1:
+                            //Add single sample parameter to pass for evalutation in SelectFinalFep
+                            finalFormalParameters.Add(flatParameters[0]);
+                            break;
+                        default:
+                            for (int j = 0; j < flatParameters.Length - 1; j++)
+                            {
+                                //Compare the type data for subsequent items
+                                if (flatParameters[j].optype != flatParameters[j + 1].optype ||
+                                    flatParameters[j].metaData.type != flatParameters[j + 1].metaData.type)
+                                {
+                                    //set function result false and exit due to disimilar function parameters
+                                    return false;
+                                }
+                            }
+
+                            //Add single sample parameter to pass for evalutation in SelectFinalFep
+                            finalFormalParameters.Add(flatParameters[0]);
```

Comment:
should this not be `finalFormalParameters.Add(flatParameters[i]);`

Reply:
At this point we don't care about which item we select.  The loop before has determined that all the `flatParameters` are the same so there I select the first item as a sample type